### ISSUE TITLE
Choose between checkboxes or multiple select for has_many

### ISSUE THREAD

**Issue by Kulgar** [blocknotes/administrate_ransack#12]
Hey, first, thanks for this gem! 

I must say it works pretty well!

I was thinking, would it be difficult to add an option to decide if we want to use checkboxes or a "select multiple" for a has_many association? 
I'd rather use the select multiple when I have a lot of records to display.

What do you think?

**Comment by blocknotes** [comment]
Hey @Kulgar.
Mmm yeah, the idea seems good.
I'll make some trials to see how it can be 👍

**Comment by Kulgar** [comment]
@blocknotes : I created a custom select field in my code to use it for my has_many associations.  

Here is my code: 

`components/_field_select.html.erb`

```rails
<% options = type.options[:collection] if type.options[:collection].present? %>
<% options ||= type.options[:collection_class].public_send(type.options[:collection_method]) if type.options[:collection_class] %>

<% field_key = model.ransackable_scopes.include?(field) ? field : "#{field}_eq" %>
<%= form.label(label) %>
<%= form.select(field_key, options || [], include_blank: true) %>
```

In my dashboard : 

```ruby
RANSACK_TYPES = {
    by_category: Administrate::Field::Select.with_options(collection_class: Category, collection_method: :as_options),
    by_author: Administrate::Field::Select.with_options(collection_class: Author, collection_method: :as_options)
}
```

In my model: 

```ruby
  scope :by_author, ->(author_id) { joins(:authors).where(authors: { id: author_id }).distinct }
  scope :by_category, ->(category_id) { joins(:categories).where(categories: { id: category_id }).distinct }

  def self.ransackable_scopes(_auth_object = nil)
    %i[by_author by_category]
  end

  def self.ransackable_scopes_skip_sanitize_args
    %i[by_author by_category]
  end
```

In fact, instead of overwriting the logic to change the field used for has many associations, I used some ransack options to create my custom scopes and used these scopes for the ransack search fields. 

You can notice I overwrote the select field component, so that I can choose between a simple frozen collection and a dynamic collection. Indeed, without the "collection_class / collection_method" options, my options in my ransack field weren't refreshing when a new category / author was created. 

I did the same for the administrate select form field ^^'

**Comment by blocknotes** [comment]
Solved in [v0.4.0](https://github.com/blocknotes/administrate_ransack/releases/tag/v0.4.0) - thanks for reporting it.
